plugin/aliases/deployment/src/main/java/io/quarkiverse/roq/plugin/aliases/deployment/RoqPluginAliasesProcessor.java

```diff
@@ -0,0 +1,113 @@
+package io.quarkiverse.roq.plugin.aliases.deployment;
+
+import static io.quarkiverse.roq.frontmatter.deployment.FrontMatterJsonData.mergeParents;
+import static io.quarkiverse.roq.frontmatter.deployment.Link.DEFAULT_PAGE_LINK_TEMPLATE;
+import static io.quarkiverse.roq.frontmatter.deployment.data.RoqFrontMatterDataProcessor.LINK_KEY;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.quarkiverse.roq.frontmatter.deployment.Link;
+import io.quarkiverse.roq.frontmatter.deployment.scan.RoqFrontMatterRawTemplateBuildItem;
+import io.quarkiverse.roq.frontmatter.runtime.RoqSiteConfig;
+import io.quarkiverse.roq.generator.deployment.items.SelectedPathBuildItem;
+import io.quarkiverse.roq.plugin.aliases.deployment.items.RoqFrontMatterAliasesBuildItem;
+import io.quarkiverse.roq.plugin.aliases.runtime.RoqFrontMatterAliasesRecorder;
+import io.quarkiverse.roq.util.PathUtils;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+import io.quarkus.vertx.http.deployment.devmode.NotFoundPageDisplayableEndpointBuildItem;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class RoqPluginAliasesProcessor {
+
+    private static final String FEATURE = "roq-plugin-aliases";
+    private static final String ALIASES_KEY = "aliases";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    public void consumeTemplates(RoqSiteConfig config, List<RoqFrontMatterRawTemplateBuildItem> rawTemplates,
```

Comment:
Ok here, you shouldn't rewrite this logic, we should add a new build-item that contains the info you need. I'll have a look..